hbase-indexer-engine/src/main/java/com/ngdata/hbaseindexer/indexer/FusionKrb5HttpClientConfigurer.java

```diff
@@ -0,0 +1,185 @@
+package com.ngdata.hbaseindexer.indexer;
+
+import com.google.common.collect.Sets;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpEntityEnclosingRequest;
+import org.apache.http.HttpException;
+import org.apache.http.HttpRequest;
+import org.apache.http.HttpRequestInterceptor;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.Credentials;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.config.AuthSchemes;
+import org.apache.http.entity.BufferedHttpEntity;
+import org.apache.http.impl.auth.SPNegoSchemeFactory;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.protocol.HttpContext;
+import org.apache.solr.client.solrj.impl.HttpClientConfigurer;
+import org.apache.solr.client.solrj.impl.HttpClientUtil;
+import org.apache.solr.common.params.SolrParams;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.CallbackHandler;
+
+import javax.security.auth.Subject;
+import javax.security.auth.callback.NameCallback;
+import javax.security.auth.callback.PasswordCallback;
+import javax.security.auth.callback.UnsupportedCallbackException;
+import javax.security.auth.kerberos.KerberosPrincipal;
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.auth.login.Configuration;
+import javax.security.auth.login.LoginContext;
+import javax.security.auth.login.LoginException;
+import java.io.IOException;
+import java.security.Principal;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class FusionKrb5HttpClientConfigurer extends HttpClientConfigurer {
+
+  private static final Logger logger = LoggerFactory.getLogger(FusionKrb5HttpClientConfigurer.class);
+  private String fusionPrincipal = null;
+
+  public static final String LOGIN_CONFIG_PROP = "java.security.auth.login.config";
+  public static final String LOGIN_APP_NAME= "fusion.jaas.appname";
+
+  private Configuration jaasConfig = null;
+
+  public Configuration getJaasConfig(){
+    return jaasConfig;
+  }
+
+  public static CloseableHttpClient createClient(String fusionPrincipal){
+    if (logger.isDebugEnabled()) {
+      System.setProperty("sun.security.krb5.debug", "true");
+    }
+    if (fusionPrincipal == null) {
+      logger.error("fusion.user (principal) must be set in order to use kerberos!");
+    }
+    HttpClientUtil.setConfigurer(new FusionKrb5HttpClientConfigurer(fusionPrincipal));
```

Comment:
Probably want to synchronize this block on the FusionKrb5HttpClientConfigurer.class object (or the HttpClientUtil class object) since you're affecting global stuff in the HttpClientUtil in a static method